src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonClaimSet.cs

```diff
@@ -24,15 +25,31 @@ internal class JsonClaimSet
 
         internal object _claimsLock = new();
         internal readonly Dictionary<string, object> _jsonClaims;
+        internal readonly Dictionary<string, (int startIndex, int length)> _jsonClaimsUtf8;
```

Comment:
can this should be in the ifdef too?

If so, could this use Range instead?

Reply:
Yes, all of this new UTF8 related code is only for .NET7+ so I'll refactor inside ifdef. Good idea, I'll see if Range works. Yea, the not ideal thing about this approach is that keeping track of indices is a bit less straightforward than just an object instance...

Reply:
Range doesn't exactly work as it's start index + end index, but we need start index + length for slicing and reading.

Reply:
Coming back to this, what about:

```cs
internal record struct MemoryOffset(int Start, int Length)
```

You can probably come up with a better name, but it is lightweight value type, and we avoid having Item1, Item2 in this code.

Reply:
This is outdated code. I added [ValuePosition](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/9d990f2a75bc3dffa6871a435630710ac8c98a15/src/Microsoft.IdentityModel.Tokens/Json/ValuePosition.cs) class. 

Also the PR description has benchmark results having this as a struct, but doesn't show perf improvement.

`record` may not be appropriate here as in an edge case when the bytes are escaped, we'll have to unescape them, and I update the ValuePosition. https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/9d990f2a75bc3dffa6871a435630710ac8c98a15/src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonClaimSet.cs#L242